File: OneCodeTeam/send-and-receive-SMTP-Mail-within-Azure-worker-22833/[C#]-send and receive SMTP Mail within Azure worker role/C#/CSAzureSendMailsByWorkerRole/WorkerRole1/WorkerRole.cs
/****************************** Module Header ******************************\
Module Name:  WorkerRole.cs
Project:      CSAzureSendMailsByWorkerRole
Copyright (c) Microsoft Corporation.
 
As you know, System.Net.Mail api can't be used in Windows Runtime application, 
However, we can create a WCF service to consume the api, and hold this service 
on Azure.

In this way, we can use this service to send email in Windows Store app. 

A workrole holds the SMTP sending mail service.
 
This source is subject to the Microsoft Public License.
See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
All other rights reserved.
 
THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\***************************************************************************/

using System.Diagnostics;
using System.Net;
using System.Threading;
using Microsoft.WindowsAzure.ServiceRuntime;
using System.ServiceModel;
using MailService;


namespace WorkerRole1
{
    public class WorkerRole : RoleEntryPoint
    {
        public override void Run()
        {
            // This is a sample worker implementation. Replace with your logic.
            while (true)
            {
                Thread.Sleep(10000);
                Trace.WriteLine("Working", "Information");
            }
        }

        public override bool OnStart()
        {
            // Set the maximum number of concurrent connections 
            ServicePointManager.DefaultConnectionLimit = 12;

            // Start the WCF host.
            ServiceHost host = new ServiceHost(typeof(MailOperationService));
            host.Open();

            return base.OnStart();
        }
    }
}

File: OneCodeTeam/send-and-receive-SMTP-Mail-within-Azure-worker-22832/[VB]-send and receive SMTP Mail within Azure worker role/VB/VBAzureSendMailsByWorkerRole/WorkerRole1/WorkerRole.vb
'****************************** Module Header ******************************\
' Module Name:  WorkerRole.vb
' Project:      VBAzureSendMailsByWorkerRole
' Copyright (c) Microsoft Corporation.
' 
' As you know, System.Net.Mail api can't be used in Windows Runtime application, 
' However, we can create a WCF service to consume the api, and hold this service 
' on Azure.
' 
' In this way, we can use this service to send email in Windows Store app. 
' 
' A workrole holds the SMTP sending mail service.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
' WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'***************************************************************************/
Imports System
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Linq
Imports System.Net
Imports System.Threading
Imports Microsoft.WindowsAzure
Imports Microsoft.WindowsAzure.Diagnostics
Imports Microsoft.WindowsAzure.ServiceRuntime
Imports Microsoft.WindowsAzure.StorageClient
Imports System.ServiceModel
Imports MailService
Public Class WorkerRole
    Inherits RoleEntryPoint

    Public Overrides Sub Run()

        ' This is a sample implementation. Replace with your logic.
        Trace.WriteLine("WorkerRole1 entry point called.", "Information")

        While (True)
            Thread.Sleep(10000)
            Trace.WriteLine("Working", "Information")
        End While

    End Sub

    Public Overrides Function OnStart() As Boolean

        ' Set the maximum number of concurrent connections 
        ServicePointManager.DefaultConnectionLimit = 12

	Dim host As New ServiceHost(GetType(MailOperationService))
	host.Open()

        ' For information on handling configuration changes
        ' see the MSDN topic at http://go.microsoft.com/fwlink/?LinkId=166357.

        Return MyBase.OnStart()

    End Function

End Class
